WikiWatershed/mmw-geoprocessing
# Implement Multi-Operation Endpoint for Subbasin Modeling

## Overview

Implements the MultiOperation outlined in #80 and #82. This accepts a JSON request specifying multipls shapes and operations, performs all of them, and returns a single JSON result.

Connects #80 

### Demo

Some initial timing:

```fish
$ for i in 1 2 3 4 5; time -p http --timeout=90 :8090/multi < examples/MultiOperationRequest.json 2>&1 > /dev/null | grep real | awk '{print $2}'; end

33.89
21.79
19.57
21.12
18.59
```

### Notes

I expect this PR to go through multiple revisions as we tweak performance.

The sample output was generated by:

```bash
$ http --timeout 90 :8090/multi < examples/MultiOperationRequest.json | jq . > examples/MultiOperationResponse.json
```

## Testing Instructions

* Checkout this branch and run `server`
* Do:

    ```bash
    http --timeout 90 :8090/multi < examples/MultiOperationRequest.json
    ```

  and see the result.
